scipy/optimize/optimize.py

```diff
@@ -1011,7 +1012,7 @@ def g(w, *args):
         return func(x0 + w*v, *args[3:])
 
     if random_projection:
-        v = random_state.randint(-1, 2, x0.shape)
+        v = random_state.binomial(1, 0.5, x0.shape)
```

Comment:
```suggestion
        v = rng_integers(random_state, 0, 2, size=x0.shape)
```

`from scipy._lib._util import rng_integers` is where you can get that from. It is designed to work with both `RandomState` and `Generator`.